reposerver/repository/repository.go

```diff
@@ -1814,6 +1817,15 @@ func getPotentiallyValidManifestFile(path string, f os.FileInfo, appPath, repoRo
 		return nil, "", nil
 	}
 
+	// Read the whole file to check whether it looks like a manifest.
+	out, rerr := utfutil.ReadFile(path, utfutil.UTF8)
+	if rerr != nil {
+		return nil, "", fmt.Errorf("failed to read %q: %w", relPath, rerr)
+	}
+	//	skip file if it contains "+argocd:skip-rendering"
+	if bytes.Contains(out, []byte("+argocd:skip-rendering")) {
```

Comment:
Extract as constant and document in go code.

I would also rename to `"+argocd:skip-file-rendering"` because it might be confusing if this is in a file containing (---) with multiple objects